### PR TITLE
feat: Add bayesian

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Support bayesian with the help of skopt.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/777)

### CODE REVIEW

File: test/scripts/v1alpha3/run-tests.sh [hunk 1]
```diff
@@ -139,5 +139,6 @@ done
 echo "Running e2e hyperopt random experiment"
 export KUBECONFIG=$HOME/.kube/config
 go run run-e2e-experiment.go ../../../examples/v1alpha3/hyperopt-random-example.yaml
+go run run-e2e-experiment.go ../../../examples/v1alpha3/skopt-bayesian-optimization-example.yaml
```
Comment (johnugeorge) [inline test/scripts/v1alpha3/run-tests.sh]: Katib config changes ?

Comment (gaocegege) [reply test/scripts/v1alpha3/run-tests.sh]: Oh forgot the change, thanks

Comment (johnugeorge) [inline test/scripts/v1alpha3/run-tests.sh]: Image has also to be replaced with CI imagee

Comment (gaocegege) [reply test/scripts/v1alpha3/run-tests.sh]: I am not sure what's your meaning. Can you explain more?

Comment (johnugeorge) [reply test/scripts/v1alpha3/run-tests.sh]: i meant this - https://github.com/kubeflow/katib/blob/master/test/scripts/v1alpha3/run-tests.sh#L89

Comment (gaocegege) [reply test/scripts/v1alpha3/run-tests.sh]: Oh yes, thanks.

**Comment by gaocegege** [comment]
Seems that we definitely need a health check in GRPC

File: pkg/controller.v1alpha3/suggestion/suggestion_controller.go [hunk 2]
```diff
@@ -201,6 +201,7 @@ func (r *ReconcileSuggestion) ReconcileSuggestion(instance *suggestionsv1alpha3.
 		}
 		msg := "Suggestion is running"
 		instance.MarkSuggestionStatusRunning(SuggestionRunningReason, msg)
+		return nil
```
Comment (johnugeorge) [inline pkg/controller.v1alpha3/suggestion/suggestion_controller.go]: Is this needed?

Comment (gaocegege) [reply pkg/controller.v1alpha3/suggestion/suggestion_controller.go]: Not needed

File: cmd/suggestion/skopt/v1alpha3/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM python:3
+
```
Comment (johnugeorge) [inline cmd/suggestion/skopt/v1alpha3/Dockerfile]: add grpc health binary?

File: pkg/controller.v1alpha3/suggestion/composer/composer.go [hunk 4]
```diff
@@ -21,7 +21,8 @@ import (
 )
 
 const (
-	defaultInitialDelaySeconds = 10
+	defaultInitialDelaySeconds = 60
```
Comment (johnugeorge) [inline pkg/controller.v1alpha3/suggestion/composer/composer.go]: do we need timeout changes again? or are they harmless?

Comment (gaocegege) [reply pkg/controller.v1alpha3/suggestion/composer/composer.go]: It does not have any side effect. I think.

Comment (gaocegege) [reply pkg/controller.v1alpha3/suggestion/composer/composer.go]: It may make suggestion not available until 60sec, should we revert to 10?

**Pull request merged**